src/main/java/org/icatproject/core/manager/search/SearchManager.java

```diff
@@ -63,47 +67,104 @@
 @Singleton
 public class SearchManager {
 
+	public static final int DEFAULT_INDEX_BATCH_SIZE = 500;
+	public static final int DEFAULT_INDEX_BATCHES_PER_TIMER = 10;
+	public static final int DEFAULT_BACKLOG_LINES_PER_TIMER = 10;
+
 	public class EnqueuedSearchRequestHandler extends TimerTask {
 
 		@Override
 		public void run() {
+			int numIndexed = 0;
+			int numBatches = 0;
+
+			while (numBatches < indexBatchesPerTimer) {
+				synchronized (queue.getReadLock()) {
+					int numIndexedInBatch = 0;
+
+					Path path;
+					try {
+						path = queue.getReadPath();
+					} catch (IcatException e) {
+						// Already logged
+						return;
+					}
+
+					if (path == null) {
+						logger.trace("No queue file available to process");
+						break;
+					}
+
+					Path dotnewPath = Paths.get(path + ".new");
+
+					logger.debug("Will attempt to process {}", path);
 
-			synchronized (queueFileLock) {
-				if (queueFile.length() != 0) {
-					logger.debug("Will attempt to process {}", queueFile);
 					StringBuilder sb = new StringBuilder("[");
-					try (BufferedReader reader = new BufferedReader(new FileReader(queueFile))) {
-						String line;
-						while ((line = reader.readLine()) != null) {
+
+					try (BufferedReader reader = Files.newBufferedReader(path)) {
+						while (numIndexedInBatch < indexBatchSize) {
+							String line = reader.readLine();
+							if (line == null) {
+								break;
+							}
+
 							if (sb.length() != 1) {
 								sb.append(',');
 							}
 							sb.append(line);
+
+							numIndexedInBatch++;
+							numIndexed++;
+						}
+
+						sb.append(']');
+
+						try (BufferedWriter writer = Files.newBufferedWriter(dotnewPath, StandardOpenOption.CREATE, StandardOpenOption.TRUNCATE_EXISTING)) {
+							String line;
+							while ((line = reader.readLine()) != null) {
+								writer.write(line);
+								writer.newLine();
+							}
+						} catch (IOException e) {
+							logger.error("Error writing to {}", dotnewPath, e);
+							return;
 						}
```

Comment:
Since this is inside the `while (numBatches < indexBatchesPerTimer)` loop, does this mean that we're potentially doing this write up to 10 tens per timer task execution? In practice there might not be much impact, but in principle it feels like were doing more IO than is needed - would there be a significant disadvantage/reason why we can't do the write/move after all the batches? I suppose it would be a bit less resilient since we have the `queue.N` file on disk with the full number of lines until the loop completes where in reality we already have either successfully committed these or written to the backLog (so we'd be double counting them in effect). If the impact of the extra writes is small then I suppose that's reason enough to do it this way?